### PR TITLE
Keep attachments on tasklist update

### LINKED ISSUE #16746: When update of a checkbox in issue, all attachments are deleted and lost.
- Gitea version (or commit ref):  1.15.0+rc3 
- Git version: builtin
- Operating system: Debian 10 / Docker
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [x] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes: https://try.gitea.io/wxiaoguang/testtest/issues/1
  - [ ] No

## Description

Steps:

1. create an issue with a checkbox and an attachment
2. click the checkbox
3. the attachment is deleted (BUG)

### PR DESCRIPTION
**Pull request by KN4CK3R**
fixes #16746

An alternative (better?) solution may be an `ignoreAttachments` parameter.
```js
await $.post(updateUrl, {
  _csrf: window.config.csrf,
  content: newContent,
  context,
  ignoreAttachments: true
});
```
```go
if !ctx.FormBool("ignoreAttachments") {
	if err := updateAttachments(comment, ctx.FormStrings("files[]")); err != nil {
		ctx.ServerError("UpdateAttachments", err)
		return
	}
}
```

### CODE REVIEW

**Comment by lunny** [comment]
> fixes #16746
> 
> An alternative (better?) solution may be an `ignoreAttachments` parameter.
> 
> ```js
> await $.post(updateUrl, {
>   _csrf: window.config.csrf,
>   content: newContent,
>   context,
>   ignoreAttachments: true
> });
> ```
> 
> ```go
> if !ctx.FormBool("ignoreAttachments") {
> 	if err := updateAttachments(comment, ctx.FormStrings("files[]")); err != nil {
> 		ctx.ServerError("UpdateAttachments", err)
> 		return
> 	}
> }
> ```

Yes, I think ignore attachment is better. Or a parameter like action=update_tasklist.

**Review by lunny — APPROVED** [review]

File: routers/web/repo/issue.go [hunk 1]
```diff
@@ -2155,6 +2148,17 @@ func UpdateCommentContent(ctx *context.Context) {
 		return
 	}
 
+	if ctx.FormString("action") == "tasklist" {
```
Comment (zeripath) [inline routers/web/repo/issue.go]:
```suggestion
	if ctx.FormString("action") == "ignoreAttachments" {
```

File: web_src/js/markup/tasklist.js [hunk 2]
```diff
@@ -46,9 +46,10 @@ export function initMarkupTasklist() {
           const {updateUrl, context} = editContentZone.dataset;
 
           await $.post(updateUrl, {
+            action: 'tasklist',
```
Comment (zeripath) [inline web_src/js/markup/tasklist.js]:
```suggestion
            action: 'ignoreAttachments',
```

**Review by zeripath — APPROVED** [review]

**Comment by wxiaoguang** [comment]
The patch seems not working ?

I just tried on https://try.gitea.io/wxiaoguang/testtest/issues/1  (1.16.0+dev-130-g0bd58d61e), `ignore_attachments=true` can be found in request, but the attachments are still deleted.

**Comment by wxiaoguang** [comment]
This PR only patches `UpdateCommentContent`.

But `UpdateIssueContent` should also be patched.

**Comment by wxiaoguang** [comment]
I proposed a new PR #16762

**Pull request merged**